controllers/template_sync.go

```diff
@@ -118,19 +121,29 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, nil
 	}
 
-	// In certain situations, we should requeue without an error, *after* other logic.
-	// Use this in non-error returns to correctly handle those situations.
-	result := reconcile.Result{}
+	// Do not exit early from the loop - store an error to return later and `continue`. Be careful
+	// not to overwrite the error in a way that it becomes nil, which would prevent a requeue.
+	// As a quirk of the error handling, only the last occurring error is "returned" by Reconcile.
+	var resultError error
 
 	// PolicyTemplates is not empty
 	// loop through policy templates
-	for _, policyT := range instance.Spec.PolicyTemplates {
+	for tIndex, policyT := range instance.Spec.PolicyTemplates {
 		object, gvk, err := unstructured.UnstructuredJSONScheme.Decode(policyT.ObjectDefinition.Raw, nil, nil)
+
+		tName := fmt.Sprintf("[template %v]", tIndex) // use a placeholder as a fallback
```

Comment:
I'm confused. Wouldn't the template be created as `[template %v]` and then fail because it's an invalid name?

I'm thinking the template name should be checked after the `Decode` error checking and then should fail if the name can't be determined.

Reply:
Good point, let me keep digging into it

Reply:
Because the name is so useful, and because it's very unclear to me when `Decode` will give an error, I put an additional failure section when the name can't be determined *before* the `Decode` error is checked.

In the case where the name can be read, but there is a Decode error, then the messages will be better. In the case where the name can't be read *because* of a Decode error, the message will just be that the name can't be found, but it will still point the user to the template index with the problem, which I expect will still be useful.

Reply:
Should've just tested this to begin with 😆 it looks like if there is a `Decode` error, then the name won't be able to be gotten. So we should handle the `Decode` error first, and just use the index instead of the name there.